### ISSUE TITLE
Summarize the bug (e.g., "Segmentation Fault while training RandLANet on S3DIS")

### ISSUE THREAD

**Issue by MSaeedMp** [isl-org/Open3D-ML#492]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D-ML/issues).
- [X] I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

I'm trying to train point transformer on a dataset based on S3DIS. 
in the last epoch of the training process, I get a value for accuracy and miou as follows:

INFO - 2022-03-11 10:36:05,257 - semantic_segmentation - Mean acc train: 0.950  eval: 0.933
INFO - 2022-03-11 10:36:05,258 - semantic_segmentation - Mean IoU train: 0.901  eval: 0.902

But when I test my model on the same sample, I get new values for accuracy and miou:
Overall Testing Accuracy : 0.6765742520464817, mIoU : 0.6481393184649503

I'm using pipeline.run_test() to test my model. 

I think the value of accuracy is being calculated differently in these stages. 

Is there any solution to this problem? 

Thank you! 

### Steps to reproduce the bug

```python
model = ml3d.models.PointTransformer(**cfg.model)
pipeline = ml3d.pipelines.SemanticSegmentation(model=model, dataset=dataset, device="gpu", **cfg.pipeline)
pipeline.run_train()

# the following lines give the value of accuracy and MIoU in the last epoch:
pipeline.metric_val.acc()
pipeline.run_test()

# this line of code gives different values of accuracy and MIoU on the same sample of data 
pipeline.run_test()
```

### Expected behavior

the value of acc and MIoU in the last epoch should be equal to the values coming from the function run_test since the same sample for validation and testing has been used. 

### Open3D, Python and System information

```markdown
- Operating system: (e.g. OSX 10.15, Ubuntu 18.04, Windows 10 64-bit)
- Python version: (e.g. Python 3.8 / output from `import sys  print(sys.version)`)
- Open3D version: (output from python: `print(open3d.__version__)`)
- System type: (x84 / arm64 / apple-silicon / jetson / rpi)
- Is this remote workstation?: yes or no
- How did you install Open3D?: (e.g. pip, conda, build from source)
- Compiler version (if built from source): (e.g. gcc 7.5, clang 7.0)
```

**Comment by maxh13** [comment]
I have a similar issue using RandLA-Net. I am running a custom dateset, with great training results ~92% accuracy and ~70ish MioU. When I attempt any form of testing, even on my validation or training data the results are not at all representative of what the training would suggest.

**Comment by vvaibhav08** [comment]
Same issue here as well. Trained RandLA-Net on a custom dataset, where the training achieves OA ~89% and mIOU ~0.82 on the train set, and OA ~75% and mIOU ~0.65 on the validation set. However, these metrics drop significantly when I use the `run_test()` function. I tested on the validation set and instead of the expected performance that I saw during training, I get OA ~35% and mIOU ~0.25.

**Comment by wmoonro** [comment]
Hello - I have a similar problem on a custom dataset, then not only with the RandLA-Net. Anyone found a likely cause for this? What is that you explored in the meantime? (I've seen that the last message dates back to Sept. 22 - did you find a way out for this?). Thank you in advance!

**Comment by borgarpa** [comment]
Same problem here using KPConv trained with PyTorch on a custom dataset!
